view/frontend/web/js/model/adyen-payment-service.js

```diff
@@ -111,14 +111,25 @@ define(
                 );
             },
 
-            donate: function (data) {
+            donate: function (data, orderId) {
+                let serviceUrl;
                 let request = {
-                    payload: JSON.stringify(data),
-                    formKey: $.mage.cookies.get('form_key')
+                    payload: JSON.stringify(data)
                 };
 
-                const serviceUrl = urlBuilder.createUrl('/internal/adyen/donations', {});
- 
+                if (customer.isLoggedIn()) {
+                    serviceUrl = urlBuilder.createUrl(
+                        '/adyen/orders/carts/mine/donations',
+                        {}
+                    );
+                } else {
+                    serviceUrl = urlBuilder.createUrl(
+                        '/adyen/orders/guest-carts/:orderId/donations', {
+                            orderId,
+                        }
+                    );
+                }
```

Comment:
We can make it more readable

```suggestion
                let path = '/adyen/orders/guest-carts/:orderId/donations'
                let urlParams = {orderId}

                if (customer.isLoggedIn()) {
                    path =  '/internal/adyen/donations'
                    urlParams = {}
                }

                const serviceUrl = urlBuilder.createUrl(path, urlParams);
```

Reply:
prefer not to introduce variables where unnecessary, but I did make it more readable